### DISCUSSION TITLE
why does getSession return null on my Ubuntu server?

### DISCUSSION THREAD

**Discussion by Rongeegee** [nextauthjs/next-auth#6632] [Help] [answered]
I get the user id from session in getServersidesprops and use the id to get the data on the server side and return the props to my page component. I have the following in my file.

```
import { getSession } from 'next-auth/react';


.....
export async function getServerSideProps(context) {
    .................
    const session = await getSession(context);
    .................
}

```

On my remote server, session is null. My server is on Ubuntu 20. However, on my localhost and the server of Onrender, session has an expected value returned from getSession method. What is reason this is happening?

the following is part of the error message

```
  url:  '<based_url>/api/auth/session',
  message: 'Unexpected token < in JSON at position 0'
```

**Comment by simonecervini** [comment, marked as answer]
As written in the documentation, `getGession()` should not be called server-side (https://next-auth.js.org/getting-started/client#getsession). The problem you are having is probably related to an avoidable `fetch` call that sometimes causes problems.

You can use instead `getServerSession()` (formerly `unstable_getServerSessions()`) to solve the problem.

**Reply by Rongeegee** [reply]
Actually, the reason for this is the nextauth_url value. On my localhost, the value of nextauth_url is http://127.0.0.1:3000. On my server, the value of nextauth_url is my domain. Once I changed the value of nextauth_url from the domain to https://127.0.0.1:80. It worked again.

**Reply by simonecervini** [reply]
It honestly seems correct to me that `NEXTAUTH_URL` is "http://localhost:3000" locally and "https://example.org" in production. It is the right configuration (https://next-auth.js.org/configuration/options#nextauth_url).

I strongly advise you to use `getServerSession()` in any case, you can find in the issues cases similar to this one with similar error message. You will probably encounter this problem eventually as well.
